package.json

```diff
@@ -13,6 +13,7 @@
     "author": "David Ryan <dryanpress@chat.wordpress.org>",
     "license": "GPL-2.0-or-later",
     "dependencies": {
+        "@newfold-labs/wp-module-ecommerce": "newfold-labs/wp-module-ecommerce#v0.2.0",
```

Comment:
What exactly is this dependency referencing? I see the module package.json points to `dist/index.js` (https://github.com/newfold-labs/wp-module-ecommerce/blob/trunk/package.json#L7), but that doesn't exist in the repo and I don't see any packages released (https://github.com/newfold-labs/wp-module-ecommerce/packages).

Reply:
npm lets you specify git sources as package url. You can refer to the [npm install](https://docs.npmjs.com/cli/v7/commands/npm-install#:~:text=npm%20install%20%3Cgithubname%3E/%3Cgithubrepo%3E%5B%23%3Ccommit%2Dish%3E%5D%3A) documentation for more details.

npm does the following things for the given package

```json
"@newfold-labs/wp-module-ecommerce": "newfold-labs/wp-module-ecommerce#v0.2.0"
```

- Search for newfold-labs/wp-module-ecommerce repo on GitHub
- Check if repo has a commit ref or tag with the value `v0.2.0`
- Download code from that commit
- Check if downloaded `package.json` file has a `prepare` script.
- Trigger the `prepare` script
- Clean up remaining files based on what `files` targets in `package.json`

In our case, the `prepare` script triggers the commands to produce the `dist/` folder which contains the needed build artifacts.